### PR TITLE
Replace jerry_value_set_abort_flag with jerry_create_abort_from_value

### PR DESCRIPTION
**Pull request by imiklos**
Replaced the function, added some tests for it. Also changed the functions order to
alphabetical.

JerryScript-DCO-1.0-Signed-off-by: Istvan Miklos imiklos2@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -904,44 +904,48 @@ jerry_is_feature_enabled (const jerry_feature_t feature) /**< feature to check *
...
- *
- * Extract the api value from an error. If the second argument is true
- * it will release the input error value.
- *
- * Note:
- *      returned value must be freed with jerry_release_value, when it is no longer needed.
- *
- * @return jerry_value_t value
- */
+* Create abort from an api value.
```
Comment (zherczeg) [inline jerry-core/api/jerry.c]: Something is wrong with the spaces.

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -951,10 +955,10 @@ jerry_create_error_from_value (jerry_value_t value, /**< api value */
   if (JERRY_UNLIKELY (ecma_is_value_error_reference (value)))
   {
     /* This is a rare case so it is optimized for
-     * binary size rather than performance. */
+    * binary size rather than performance. */
```
Comment (zherczeg) [inline jerry-core/api/jerry.c]: Ditto. Please check the whole patch.

File: tests/unit-core/test-api-set-and-clear-error-flag.c [hunk 3]
```diff
@@ -130,9 +185,21 @@ main (void)
   error = jerry_create_error_from_value (error, true);
   jerry_release_value (error);
 
+  value = jerry_create_number (42);
+  error = jerry_create_abort_from_value (value, true);
+  error = jerry_create_abort_from_value (error, true);
+  jerry_release_value (error);
+
+
```
Comment (zherczeg) [inline tests/unit-core/test-api-set-and-clear-error-flag.c]: One newline here.

Comment (imiklos) [reply tests/unit-core/test-api-set-and-clear-error-flag.c]: Didn't see that, thank you. I updated the PR.

**Pull request merged**